src/app/data-model-provider/Provider.h

```diff
@@ -125,7 +129,7 @@ class Provider : public ProviderMetadataTree
                                                             CommandHandler * handler) = 0;
 
 protected:
-    InteractionModelContext mContext = {};
+    std::optional<InteractionModelContext> mContext;
```

Comment:
This could use a comment explaining why this thing is optional.

Reply:
Added a comment. Also made `CurrentContext` return an optional, so that callers need to handle it missing (and added handling inside the codegen datamodel provider logic).

The usage of this seems quite limited ... I wonder if this should be really part of the public API or not since this was intended for transition only and clusters are the ones that keep context data instead of the provider.